### PR TITLE
feature: prisma-select transform option suggestion

### PR DESCRIPTION
**Pull request by CreeJee**
## Background 
I need to orderBy for relational fields. 
And sometimes we need to add for optional select object "complex query prisma object".

## what changed
- add option for transform,
    - "find": search graphql field name callback
    - "update": when found search callback that change property
- update deps for latest
- fix some tests and build working on windows os 
- change dmmf snapshot 

Sorry for my bad English!

### CODE REVIEW

**Comment by AhmedElywa** [comment]
It's a very big pull request give me some time to review this code

File: packages/create/package.json [hunk 1]
```diff
@@ -37,7 +38,7 @@
     "cross-spawn": "^7.0.3",
     "rimraf": "^3.0.2",
     "tslib": "^2.3.1",
-    "typescript": "^5.0.2",
+    "typescript": "^5.x",
```
Comment (AhmedElywa) [inline packages/create/package.json:41]:
```suggestion
    "typescript": "^5",
```

File: packages/display/package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
   "devDependencies": {
     "@types/debug": "^4.1.7",
     "@types/node": "18.15.3",
-    "typescript": "^5.0.2"
+    "typescript": "^5.x",
```
Comment (AhmedElywa) [inline packages/display/package.json:37]:
```suggestion
    "typescript": "^5",
```

File: packages/generator/package.json [hunk 3]
```diff
@@ -28,7 +28,8 @@
     "@types/node": "18.15.3",
     "@types/pluralize": "^0.0.29",
     "@types/prettier": "^2.7.2",
-    "typescript": "^5.0.2"
+    "typescript": "^5.x",
```
Comment (AhmedElywa) [inline packages/generator/package.json:31]:
```suggestion
    "typescript": "^5",
```

File: packages/schema/package.json [hunk 4]
```diff
@@ -25,7 +25,8 @@
     "@paljs/utils": "workspace:*"
   },
   "devDependencies": {
-    "typescript": "^5.0.2"
+    "typescript": "^5.x",
```
Comment (AhmedElywa) [inline packages/schema/package.json:28]:
```suggestion
    "typescript": "^5",
```

File: packages/utils/package.json [hunk 5]
```diff
@@ -14,13 +14,14 @@
...
     "build": "pnpm run clean && tsc -p tsconfig.build.json",
-    "clean": "rm -rf dist node_modules/.cache"
+    "clean": "shx rm -rf dist node_modules/.cache"
   },
   "publishConfig": {
     "access": "public"
   },
   "devDependencies": {
-    "typescript": "^5.0.2"
+    "typescript": "^5.x",
```
Comment (AhmedElywa) [inline packages/utils/package.json:23]:
```suggestion
    "typescript": "^5",
```

**Review by AhmedElywa — CHANGES_REQUESTED** [review]

File: packages/plugins/src/select.ts [hunk 6]
```diff
@@ -1,7 +1,77 @@
...
+   *             }
+   *           }
+   *         }
+   *       }
+   *     },
+   *     // Add more transformation objects as needed
+   *   ]
+   * }
+   */
+  transform?: FindAndTransformOption[];
```
[the remark this answers is not in the document]
Comment (CreeJee) [reply packages/plugins/src/select.ts:73]: Okay 👍

**Comment by AhmedElywa** [comment]
We have a big of changes in our project to support prisma v5 
if you want to merge this, please update it

**Comment by CreeJee** [comment]
okay...
but i have a no time to merge suggestion...
but i'll try it

**Comment by AhmedElywa** [comment]
Thanks, i closed it for now

**Pull request closed without merging**